packages/server/src/adapters/zodFileSchema.ts

```diff
@@ -0,0 +1,76 @@
+import type { Readable } from 'stream';
+import z from 'zod';
+
+function isObject(value: unknown): value is Record<string, unknown> {
+  // check that value is object
+  return !!value && !Array.isArray(value) && typeof value === 'object';
+}
+
+const readableEsqueSchema = z.custom<Readable>((value) => {
+  // check if file is is a Readable without importing Readable
+  if (!isObject(value)) {
+    return false;
+  }
+  if (typeof (value as unknown as Readable).on === 'function') {
+    return true;
+  }
+
+  return false;
+});
+
+/**
+ * This is a hack to check if a value is a File in the browser
+ */
+const fileEsqueSchema = z.custom<File>((value) => {
+  if (!isObject(value)) {
+    return false;
+  }
+  if (typeof (value as unknown as File).type === 'string') {
+    return true;
+  }
+  return false;
+});
+
+/**
+ * This is a hack to check if a value is a FileList in the browser
```

Comment:
`FileList` is defined in browser so you could just do `instanceof` no? Is this supposed to be "Nodejs"?

Reply:
The problem is highlighted here: https://stackblitz.com/edit/node-ry4feb?file=index.ts

An empty `FileList` should be treated the same as `undefined`:

![CleanShot 2023-03-04 at 23 58 42@2x](https://user-images.githubusercontent.com/459267/222932531-c73dbd5f-7075-45a5-a811-acd0eef0a7c2.png)


Is this even possible?